src/plugins/discover/public/application/apps/main/components/sidebar/change_indexpattern.tsx

```diff
@@ -86,7 +86,9 @@ export function ChangeIndexPattern({
             const choice = (choices.find(({ checked }) => checked) as unknown) as {
               value: string;
             };
-            onChangeIndexPattern(choice.value);
+            if (choice.value !== indexPatternId) {
+              onChangeIndexPattern(choice.value);
```

Comment:
It might be redundant, but suppose it make sense on the other hand to disable selected index pattern option if we can

Reply:
synced offline, and decided to keep it that way, because when it's disabled, some functional tests might no be able to select the same index pattern